### PR TITLE
Adds array handling to `forc fmt`

### LINKED ISSUE #726: `forc fmt` doesn't format arrays very well.
Here's what it did to a recent PR of mine:
```diff
diff --git a/src/b512.sw b/src/b512.sw
index 2bcf573..ff5f664 100644
--- a/src/b512.sw
+++ b/src/b512.sw
@@ -4,7 +4,8 @@ library b512;
 /// Stores two b256s in contiguous memory.
 /// Guaranteed to be contiguous for use with ec-recover: std::ecr::ec_recover().
 pub struct B512 {
-    bytes: [b256; 2],
+    bytes: [b256;
+    2],
 }

 // @todo use generic form when possible
@@ -19,7 +20,8 @@ pub trait From {
 impl From for B512 {
     fn from(h: b256, l: b256) -> B512 {
         B512 {
-            bytes: [h, l],
+            bytes: [h,
+            l],
         }
     }
 }
@@ -30,7 +32,7 @@ impl B512 {
     fn new() -> B512 {
         B512 {
             bytes: [0x0000000000000000000000000000000000000000000000000000000000000000,
-                    0x0000000000000000000000000000000000000000000000000000000000000000],
+            0x0000000000000000000000000000000000000000000000000000000000000000],
         }
     }
 }
```

### PR DESCRIPTION
**Pull request by eureka-cpu**
Closes #726 

- [x] handles collections
- [x] handles nested collections
- [ ] special formatting for large array lengths
- [ ] implement max-width for code lines
- [ ] add tests to `./sway-fmt/test`

### CODE REVIEW

**Comment by leviathanbeak** [comment]
left a comment at #726

File: sway-fmt/src/code_builder.rs [hunk 1]
```diff
@@ -120,6 +119,57 @@ impl CodeBuilder {
                             }
                         }
 
+                        '[' => {
+                            if code_line.is_collection() {
+                                if code_line.text.len() >= 60 {
```
Comment (adlerjohn) [inline sway-fmt/src/code_builder.rs:124]: Turn this magic `60` into a constant

Comment (eureka-cpu) [reply sway-fmt/src/code_builder.rs:124]: Will do! (:

**Comment by eureka-cpu** [comment]
New parser is up, and the architecture for the formatter is going to change soon so I'll close this for now and use what I'll need from it for the new incoming formatter.

**Pull request closed without merging**